### PR TITLE
236 remove split on comma

### PR DESCRIPTION
**Pull request by castaway**
All "TO" fields updated (and tests passing hopefully)

Also done: CC + BCC fields

### CODE REVIEW

File: src/app/compose/draftdesk.service.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ export class DraftFormModel {
         const ret = new DraftFormModel();
         ret.from = fromAddress.email;
         ret.mid = draftId;
-        ret.to = to;
+        ret.to = to ? MailAddressInfo.parse(to) : null;
```
Comment (tadzik) [inline src/app/compose/draftdesk.service.ts:61]: If we declare it as an array, and the default value is `[]` maybe we should default to `[]` over `null` here too?

Comment (castaway) [reply src/app/compose/draftdesk.service.ts:61]: Oops, yes seems saner

File: src/app/compose/draftdesk.service.spec.ts [hunk 2]
```diff
@@ -33,56 +33,209 @@ describe('DraftDesk', () => {
...
-        draft = DraftFormModel.reply({
+        done();
+    });
+    it('Reply: Address with MAI', (done) => {
+        console.log('Reply test: Address with MAI');
+        const mailDate = new Date(2017, 6, 1);
+
+        const timezoneOffset: number = mailDate.getTimezoneOffset();
+
+        const timezoneOffsetString: string = 'GMT' + (timezoneOffset <= 0 ? '+' : '-') +
```
Comment (tadzik) [inline src/app/compose/draftdesk.service.spec.ts:152]: This piece is duplicated in this file, could probably be refactored into a function.

I wonder, though, if we really need all this computing? Do the tests care for something more than "the timezone is the same before and after?"

**Comment by tadzik** [comment]
Found an issue:

1) Open the CC form
2) Enter some email address
3) Close the CC form with the X
4) Open the CC form again
5) The previous email address is still there

That's because the X still resets the form contents, not the model itself.

File: src/app/compose/draftdesk.service.ts [hunk 3]
```diff
@@ -112,7 +119,7 @@ export class DraftFormModel {
 
         if (!useHTML && mailObj.rawtext) {
             ret.msg_body = '\n' + mailDate.toISOString().substr(0, 'yyyy-MM-ddTHH:mm'.length).replace('T', ' ') + ' ' +
-                timezoneOffsetString + ' ' + sender + ':\n' +
+                timezoneOffsetString + ' ' + sender[0].nameAndAddress + ':\n' +
```
[the remark this answers is not in the document]
Comment (castaway) [reply src/app/compose/draftdesk.service.ts:122]: If sender is empty / not an array then several other things will break too - we dont currently check any of the incoming call arguments for sanity in this code..

File: src/app/compose/draftdesk.service.spec.ts (hunk 2 above)
Comment (castaway) [reply src/app/compose/draftdesk.service.spec.ts:152]: Good point, I just duplicated what was already there..

**Comment by castaway** [comment]
> Found an issue:
> 
> 1. Open the CC form
> 2. Enter some email address
> 3. Close the CC form with the X
> 4. Open the CC form again
> 5. The previous email address is still there
> 
> That's because the X still resets the form contents, not the model itself.

Oops, now fixed - even less use of / reliance on the formControl objects (wonder if I can delete em yet)

**Pull request merged**